covid_moonshot/analysis/plots.py

```diff
@@ -0,0 +1,448 @@
+    solvent_delta_fs, solvent_delta_f_errs : list of float
+        Free energies and errors for the solvent; one of each per gen (in kT)
+    binding_delta_f, binding_delta_f_err : float
+        Binding free energy and error, estimated using data for all gens (in kT)
+    n_devs_bounds : float
+        Number of standard deviations for drawing bounds
+
+    Returns
+    -------
+    Figure
+        Figure containing the plot
+    """
+
+    fig, (ax1, ax2) = plt.subplots(nrows=2, sharex=True)
+
+    DDG = np.array(solvent_delta_fs) - np.array(complex_delta_fs)
+    DDG_kC = (DDG * KT).value_in_unit(unit.kilocalories_per_mole)
+
+    DDG_err = np.sqrt(
+        np.array(solvent_delta_f_errs) ** 2 + np.array(complex_delta_f_errs) ** 2
+    )
+    DDG_err_kC = (DDG_err * KT).value_in_unit(unit.kilocalories_per_mole)
+
+    ax1.scatter(gens, DDG_kC, color="green", label="binding")
+    ax1.vlines(
+        gens,
+        DDG_kC - DDG_err_kC * n_devs_bounds,
+        DDG_kC + DDG_err_kC * n_devs_bounds,
+        color="green",
+    )
+
+    for label, (delta_fs, delta_f_errs), color in [
+        ("solvent", (solvent_delta_fs, solvent_delta_f_errs), "blue"),
+        ("complex", (complex_delta_fs, complex_delta_f_errs), "red"),
+    ]:
+
+        delta_fs_kC = (np.array(delta_fs) * KT).value_in_unit(
+            unit.kilocalories_per_mole
+        )
+        delta_f_errs_kC = (np.array(delta_f_errs) * KT).value_in_unit(
+            unit.kilocalories_per_mole
+        )
+
+        ax2.scatter(gens, delta_fs_kC, color=color, label=label)
+        ax2.vlines(
+            gens,
+            delta_fs_kC - delta_f_errs_kC * n_devs_bounds,
+            delta_fs_kC + delta_f_errs_kC * n_devs_bounds,
+            color=color,
+        )
+
+    ax1.hlines(
+        (binding_delta_f * KT).value_in_unit(unit.kilocalories_per_mole),
+        0,
+        max(gens),
+        color="green",
+        linestyle=":",
+        label="binding (all GENS)",
+    )
+    ax1.fill_between(
+        [0, max(gens)],
+        ((binding_delta_f - binding_delta_f_err * n_devs_bounds) * KT).value_in_unit(
+            unit.kilocalories_per_mole
+        ),
+        ((binding_delta_f + binding_delta_f_err * n_devs_bounds) * KT).value_in_unit(
+            unit.kilocalories_per_mole
+        ),
+        alpha=0.2,
+        color="green",
+    )
+
+    ax1.set_xticks([gen for gen in range(0, max(gens) + 1)])
+    ax2.set_xlabel("GEN")
+    ax1.legend()
+    ax2.legend()
+
+    for ax in [ax1, ax2]:
+        ax.set_ylabel("Rel. $\Delta F$ /" + r" kcal mol${^-1}$")
+
+    return fig
+
+
+def plot_cumulative_distributions(
+    affinities: List[float],
+    minimum: Optional[float] = None,
+    maximum: float = 5,
+    cmap: str = "PiYG",
+    n_bins: int = 100,
+    markers: List[float] = [-2, -1, 0, 1, 2],
+) -> None:
+    """
+    Plot cumulative distribution of ligand affinities
+
+    Parameters
+    ----------
+    affinities : list of float
```

Comment:
What units are these supposed to be in?
Let's also be a bit careful to call then "delta_f" or "free_energies" since the relationship between the free energy and affinity is
```
affinity = exp(-free_energy / kT) * (1 Molar)
```
We generally don't want to use affinities in our reports for internal use.

Reply:
Ah! I wasn't aware of this convention. Makes sense. I replaced all occurrences of "affinity" with "delta_f" or "free energy" in 9eddd3d, 56cfc0b